core/base/mtx_io.cpp

```diff
@@ -341,7 +342,8 @@ class mtx_io {
         size_type get_reservation_size(size_type num_rows, size_type num_cols,
                                        size_type num_nonzeros) const override
         {
-            return 2 * num_nonzeros - max(num_rows, num_cols);
+            return 2 * num_nonzeros -
+                   min(2 * num_nonzeros, max(num_rows, num_cols));
```

Comment:
```suggestion
            return 2 * num_nonzeros;
```
the reservation is 0 if 2#nnz is smaller than num_rows or num_cols.
If the reservation required to be large enough, it might only use 2 nnz (the worst case: no diagonal)

Reply:
It's not required to be large enough, we only want to allocate enough memory for the common cases. For common, non-hypersparse symmetric matrices, the diagonal is full, and this gives a perfect estimate.